src/libexpr/primops.cc

```diff
@@ -77,18 +76,55 @@ StringMap EvalState::realiseContext(const NixStringContext & context)
             "cannot build '%1%' during evaluation because the option 'allow-import-from-derivation' is disabled",
             store->printStorePath(drvs.begin()->drvPath)));
 
+    if (evalSettings.logImportFromDerivation) {
+         for (auto & drv : drvs) {
+              auto pos2 = positions[pos];
+              std::ostringstream str;
+              ((std::shared_ptr<AbstractPos>) pos2)->print(str);
+              Activity act{
+                   *logger, lvlInfo, actFromDerivation,
+                   fmt("Derivation %s Output %s: importing from derivation %s via %s",
+                       store->printStorePath(drv.drvPath), drv.output, positions[pos], reason),
+                   {
+                        Logger::Field{store->printStorePath(drv.drvPath)},
+                        Logger::Field{drv.output},
+                        Logger::Field{
```

Comment:
Do we want this split out or do we want to construct the Field in a way where it's the same as `positions[pos]`?

Reply:
I got keep it this way, so maximum structure for the an external application. And pull out the `AbstactPos -> Fields` conversion to a separate function.